butterknife/src/main/java/butterknife/ButterKnife.java

```diff
@@ -114,24 +115,14 @@ private ButterKnife() {
     };
 
     private static <T> T[] filterNull(T[] views) {
-      int newSize = views.length;
-      for (T view : views) {
-        if (view == null) {
-          newSize -= 1;
-        }
-      }
-      if (newSize == views.length) {
-        return views;
-      }
-      //noinspection unchecked
-      T[] newViews = (T[]) new Object[newSize];
-      int nextIndex = 0;
-      for (T view : views) {
+      int end = 0;
+      for (int i = 0; i < views.length; i++) {
+        T view = views[i];
         if (view != null) {
-          newViews[nextIndex++] = view;
+          views[end++] = view;
```

Comment:
Well, a one liner change to the existing implementation would be to change 127 to what `Arrays.copyOfRange` does under the hood:

```java
T[] newViews = Array.newInstance(views.getClass().getComponentType(), newSize);
```

When I think about it, I don't know which option would really be better in this case though. I'd speculate that the average case (where most, if not all the views are non-null) is faster in the original implementation using `Array.newInstance`. I'm not sure what point the `Arrays.copyOfRange` implementation would become faster, but honestly think either case would be fine perf-wise. Anyone with enough views to cause a noticeable difference probably has bigger problems.